network-api/networkapi/wagtailpages/pagemodels/libraries/rcc/library_page.py

```diff
@@ -65,31 +67,33 @@ class RCCLibraryPage(BasePage):
     ]
 
     def get_context(self, request: "http.HttpRequest") -> "django_template.Context":
-        #     search_query: str = request.GET.get("search", "")
+        search_query: str = request.GET.get("search", "")
         sort_value: str = request.GET.get("sort", "")
         sort: constants.SortOption = constants.SORT_CHOICES.get(sort_value, constants.SORT_NEWEST_FIRST)
 
-        #     filter_form = RCCLibraryPageFilterForm(request.GET, label_suffix="")
-        #     if not filter_form.is_valid():
-        #         # If the form is not valid, we will not filter by any of the values.
-        #         # This will result in all articles being displayed.
-        #         filtered_author_ids: list[int] = []
-        #         filtered_content_type_ids: list[int] = []
-        #         filtered_curricular_area_ids: list[int] = []
-        #         filtered_topic_ids: list[int] = []
-
-        #     filtered_author_ids = filter_form.cleaned_data["contributors"]
-        #     filtered_content_type_ids = filter_form.cleaned_data["content_types"]
-        #     filtered_curricular_area_ids = filter_form.cleaned_data["curricular_areas"]
-        #     filtered_topic_ids = filter_form.cleaned_data["topics"]
-
-        searched_and_filtered_rcc_detail_pages = self._get_rcc_detail_pages(sort=sort)
-        # search=search_query,
-        # sort=sort,
-        # author_profile_ids=filtered_author_ids,
-        # content_type_ids=filtered_content_type_ids,
-        # curricular_area_ids=filtered_curricular_area_ids,
-        # topic_ids=filtered_topic_ids,
+        filter_form = RCCLibraryPageFilterForm(request.GET, label_suffix="")
+        if not filter_form.is_valid():
+            # If the form is not valid, we will not filter by any of the values.
+            # This will result in all articles being displayed.
+            filtered_author_ids: list[int] = []
+            filtered_content_type_ids: list[int] = []
+            filtered_curricular_area_ids: list[int] = []
+            filtered_topic_ids: list[int] = []
```

Comment:
I don't think these veriable assignments are needed as they are immediately re-assigned on line 83

Reply:
It works, although it's not the most readable code. The equivalent is:

```python
        if filter_form.is_valid():
            filtered_author_ids = filter_form.cleaned_data["contributors"]
            filtered_content_type_ids = filter_form.cleaned_data["content_types"]
            filtered_curricular_area_ids = filter_form.cleaned_data["curricular_areas"]
            filtered_topic_ids = filter_form.cleaned_data["topics"]
        else:
            # If the form is not valid, we will not filter by any of the values.
            # This will result in all articles being displayed.
            filtered_author_ids: list[int] = []
            filtered_content_type_ids: list[int] = []
            filtered_curricular_area_ids: list[int] = []
            filtered_topic_ids: list[int] = []
```

I can replace it by that form, to make it more readable.